src/combat.ts

```diff
@@ -11,14 +10,18 @@ export default class Macro extends StrictMacro {
         `!mpbelow ${mpCost($skill`Stuffed Mortar Shell`)}`,
         Macro.trySkill($skill`Stuffed Mortar Shell`)
       );
+
+    const whileCondition = (sk: Skill): [string, Macro] => {
+      return [`!mpbelow ${sk} && hasskill ${toInt(sk)}`, Macro.skill(sk)];
+    };
+
     return (useCinch ? macroHead.trySkill($skill`Cincho: Confetti Extravaganza`) : macroHead)
-      .while_(
-        `!mpbelow ${mpCost($skill`Saucegeyser`)} && hasskill ${toInt($skill`Saucegeyser`)}`,
-        Macro.skill($skill`Saucegeyser`)
-      )
-      .while_(
-        `!mpbelow ${mpCost($skill`Saucestorm`)} && hasskill ${toInt($skill`Saucestorm`)}`,
-        Macro.skill($skill`Saucestorm`)
+      .externalIf(
+        mainStat === $stat`Muscle`,
+        Macro.while_(...whileCondition($skill`Lunging Thrust-Smack`)),
+        Macro.while_(...whileCondition($skill`Saucegeyser`)).while_(
+          ...whileCondition($skill`Saucestorm`)
+        )
```

Comment:
Unfortunately, this refactoring of `...whileCondition` does not seem to work, so we will have to manually hardcode each `while_` condition